llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp

```diff
@@ -2016,7 +2016,8 @@ SDValue VectorLegalizer::UnrollVSETCC(SDNode *Node) {
                          TLI.getSetCCResultType(DAG.getDataLayout(),
                                                 *DAG.getContext(), TmpEltVT),
                          LHSElem, RHSElem, CC);
-    Ops[i] = DAG.getSelect(dl, EltVT, Ops[i], DAG.getAllOnesConstant(dl, EltVT),
+    Ops[i] = DAG.getSelect(dl, EltVT, Ops[i],
+                           DAG.getBoolConstant(true, dl, EltVT, VT),
```

Comment:
This isn't correct as we might have a different getBooleanContents result for EltVT vs VT

Reply:
> This isn't correct as we might have a different getBooleanContents result for EltVT vs VT

I suppose we should actually use `VT` for `getBooleanContents` result. Because we are doing some "Unrolling" and after that that the Ops are constructed in a vector (by `BUILD_VECTOR`). So using `VT` (the vector type) it the correct fix I suppose?

Reply:
But this isn't NFC and we definitely need test coverage

Reply:
Instead of using the select on constant, why not use getBoolExt

Reply:
@inclyc Maybe try this:
```
    Ops[i] = DAG.getNode(ISD::SETCC, dl, MVT::i1, LHSElem, RHSElem, CC);
    Ops[i] = DAG.getBoolExtOrTrunc(Ops[i], dl, EltVT, VT);
```

Reply:
Thanks, updated! @arsenm @RKSimon